lambdas/call-dcmaw-async-cri/src/test/java/uk/gov/di/ipv/core/calldcmawasynccri/pact/dcmawasynccri/ContractTest.java

```diff
@@ -270,6 +270,51 @@ public RequestResponsePact invalidAccessTokenReturns400(PactDslWithProvider buil
         assertEquals(HTTPResponse.SC_SERVER_ERROR, exception.getHttpStatusCode());
     }
 
+    @Pact(provider = "DcmawAsyncCriProvider", consumer = "IpvCoreBack")
+    public RequestResponsePact missingAccessTokenReturns401(PactDslWithProvider builder)
+            throws Exception {
+        return builder.given("access token is missing")
+                .uponReceiving("Valid credential request")
+                .path("/async/credential")
+                .method("POST")
+                .body(OBJECT_MAPPER.writeValueAsString(getCredentialRequestBody(SUBJECT_ID)))
+                .headers("Content-Type", "application/json")
+                .willRespondWith()
+                .status(401)
+                .toPact();
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "missingAccessTokenReturns401")
+    void
+            fetchVerifiableCredential_whenCalledAgainstDcmawAsyncCriWithMissingAccessToken_throwsAnException(
+                    MockServer mockServer) throws URISyntaxException {
+        // Arrange
+        var credentialIssuerConfig = getMockCredentialIssuerConfig(mockServer);
+        when(mockConfigService.getOauthCriConfig(CRI_OAUTH_SESSION_ITEM))
+                .thenReturn(credentialIssuerConfig);
+
+        var underTest =
+                new CriApiService(
+                        mockConfigService, mockSignerFactory, mockSecureTokenHelper, CURRENT_TIME);
+
+        // Act
+        CriApiException exception =
+                assertThrows(
+                        CriApiException.class,
+                        () ->
+                                underTest.fetchVerifiableCredential(
+                                        new BearerAccessToken("missingAccessToken"),
```

Comment:
@MikeCollingwood - we do return a 401 in the scenario where the request to `async/credential` doesn't contain:

- An Authorization header
- A bearer token in the format 'Bearer xyz'

Adding this scenario just to document it. Happy to remove it.

I'm not sure it's possible to test this without passing in an argument to the `new BearerAccessToken` function here - it looks a bit strange so happy to remove/adjust.

Reply:
I've stubbed the `BearerAccessToken` in a recent commit to return null on `.toAuthorizationHeader()`